File: backend/app/schemas/food_truck.py
from pydantic import BaseModel, Field


class FoodTruck(BaseModel):
    id: str = Field(alias="objectid")
    name: str = Field(alias="applicant")
    location_description: str | None = Field(
        default=None,
        alias="locationdescription",
    )
    address: str | None = None
    permit: str | None = None
    food_items: str | None = Field(
        default=None,
        alias="fooditems",
    )
    latitude: float
    longitude: float
    days_hours: str | None = Field(
        default=None,
        alias="dayshours",
    )

    model_config = {
        "populate_by_name": True,
    }

File: frontend/src/app/core/models/food-truck.ts
export interface FoodTruck {
  id: string;
  name: string;

  locationDescription?: string;
  address?: string;
  permit?: string;
  foodItems?: string;

  latitude: number;
  longitude: number;

  daysHours?: string;
}

export interface FoodTruckApiResponse {
  objectid: string | number;
  applicant?: string;

  locationdescription?: string;
  address?: string;
  permit?: string;
  fooditems?: string;

  latitude: string | number;
  longitude: string | number;

  dayshours?: string | null;
}